### ISSUE TITLE
inline/dce issue

### ISSUE THREAD

**Issue by azrafe7** [HaxeFoundation/haxe#6721]
This is much probably unwanted behaviour from a user point of view, but not sure if it's expected or not.

```haxe
  static function main():Void {
    log("not dce'd" + Int64.ofInt(12));
  }
  
  inline static public function log(s:String):Void {
    return ;
  }
```

Compiling to js (for example) generates:

```js
IssueDce.main = function() {
  var this1 = new haxe__$Int64__$_$_$Int64(0,12);
  var s = "not dce'd" + haxe__$Int64_Int64_$Impl_$.toString(this1);
};
```

(using with haxe 4.0.0 (git build development @ c76e38a), with `-dce full` and `-D analyzer`)

Is there a way already in place to avoid evaluating the temp vars (maybe some meta I don't know of)?

**Comment by Simn** [comment]
This is not unwanted behavior because it follows the normal semantics of calls. The function being inline should not influence that.

This particular case would be fixed by adding `@:pure` to Int64.toString. That's not technically correct though because it uses divMod, and divMod can throw an exception. So I'm not sure if we should do that...

**Comment by azrafe7** [comment]
My use case is very similar to what `--no-traces` does: strip out all call occurrences from the output and don't bother with eventual passed args (is that achievable?). My understanding is that it has some special support in the compiler but may be wrong.

I'm reluctant to override `haxe.Log.trace()` (I've played a bit with it), as my only intent is for debug use in a lib and I don't want to force it on the user (and risk conflicting with other libs).

I've tried adding `@:pure` to `log()`, but as you point out it doesn't work, and rightly so!

What are my options here? (macro I guess, but any easier way?)
